llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -15887,7 +15889,14 @@ static bool isCopyFromRegForI1Return(const SDNode *N) {
   SDNode *N2 = N1->getOperand(0).getNode();
   if (N2->getOpcode() != ISD::CopyFromReg)
     return false;
-  SDNode *N3 = N2->getOperand(0).getNode();
+
```

Comment:
This whole function shouldn't be necessary. You can't necessarily identify it's from a return context, but you shouldn't need to reverse engineer this

Reply:
Without this, the assert `assert(Reg == FLI->DemoteRegister || isCopyFromRegOfInlineAsm(N)` in `SITargetLowering::isSDNodeSourceOfDivergence()` would trigger in debug build. As I mentioned earlier, an alternative is to use the phys reg from the original CopyFromReg instead of a virtual reg in the new CopyFromReg.

Reply:
Instead of this, you need to adjust isArgPassedInSGPR to account for the i1 case

Reply:
Although this case does break the "SGPR = uniform" based logic spread through the DAG

Reply:
So you also have to update isSourceOfDivergence, on top of isArgPassedinSGPR

Reply:
ping here, if UniformityAnalysis was correct I don't think you need this

Reply:
Pls clarify. Are you saying the entire function `isCopyFromRegForI1Return()` is not needed? Also, you said "if UniformityAnalysis was correct". So it is not correct and needs fix?

Reply:
Yes, this whole function should be unnecessary. isArgPassedInSGPR needs to be updated, and the use for uniformity needs to be updated to account for this non-uniform, but also in SGPR case